pkg/daemon/server/service/isbs_query_service.go

```diff
@@ -12,15 +16,25 @@ import (
 	"github.com/numaproj/numaflow/pkg/shared/logging"
 )
 
+// HTTPClient interface for the GET call to metrics endpoint. Had to add this an interface for testing
+type HTTPClient interface {
+	Get(url string) (*http.Response, error)
+}
+
 type isbSvcQueryService struct {
-	client   isbsvc.ISBService
-	pipeline *v1alpha1.Pipeline
+	client     isbsvc.ISBService
+	pipeline   *v1alpha1.Pipeline
+	httpClient HTTPClient
 }
 
+// NewISBSvcQueryService returns a new instance of isbSvcQueryService
 func NewISBSvcQueryService(client isbsvc.ISBService, pipeline *v1alpha1.Pipeline) *isbSvcQueryService {
```

Comment:
Should we now call this something else, as it is no longer isbs query service.

Reply:
That's also what I'm thinking.

Reply:
Changed it to pipeline_metrics_query_service